### PR TITLE
Change Carbon DB into Registry DB and UM DB

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: pattern-1/bosh-release/config/blobs.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+jdbcdrivers/mssql-jdbc-7.0.0.jre8.jar:
```
Comment (VimukthiPerera) [inline pattern-1/bosh-release/config/blobs.yml]: Shall we keep this file empty as these SHA values change every time the BOSH release is created?

File: pattern-2/bosh-release/config/blobs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+jdbcdrivers/mssql-jdbc-7.0.0.jre8.jar:
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/config/blobs.yml]: Shall we keep this file empty as well?

File: pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb [hunk 3]
```diff
@@ -65,7 +65,7 @@ dpkg -s nfs-common >/dev/null 2>&1
 if [ $? -ne 0 ] ; then
   log_debug "Installing NFS Client"
   sudo apt-get update
-  sudo apt-get install nfs-common
+  sudo apt-get install nfs-common -y
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb:68]: This is needed if we are changing the stemcell. But the stemcell in the tile.yaml is still the same.

File: pattern-1/bosh-release/create.sh [hunk 4]
```diff
@@ -97,6 +97,9 @@ if [ ! -x "$(command -v bosh)" ]; then
 fi
 
 # move to the deployment directory
+if [ ! -d ../${deployment} ]; then
```
Comment (VimukthiPerera) [inline pattern-1/bosh-release/create.sh]: Shall we add a .gitkeep file to the deployment folder? Then we can update the README with the structure of the BOSH release.

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

**Comment by VimukthiPerera** [comment]
Any reason to remove the .gitkeep file in the  pattern-1/tile/cache/ directory?

File: pattern-2/bosh-release/jobs/gateway/templates/ctl.erb [hunk 5]
```diff
@@ -67,7 +67,7 @@ dpkg -s nfs-common >/dev/null 2>&1
 if [ $? -ne 0 ] ; then
   log_debug "Installing NFS Client"
   sudo apt-get update
-  sudo apt-get install nfs-common
+  sudo apt-get install nfs-common -y
```
Comment (VimukthiPerera) [inline pattern-2/bosh-release/jobs/gateway/templates/ctl.erb:70]: We need this option only if we change the stemcell. Shall we update the stemcell as well?

File: pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb (hunk 3 above)
Comment (Savidude) [reply pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb:68]: Shall we change the stemcell version as well?

File: pattern-2/tile/tile.yml [hunk 6]
```diff
@@ -22,19 +22,19 @@ label: WSO2 API Manager 2.6.0
...
-  version: '3541'
+  version: '170.15'
 
 forms:
-- name: carbon_db
-  label: Carbon DB connection information
-  description: WSO2 API Manager - Carbon Database connection information
+- name: reg_db
+  label: Registry DB connection information
+  description: WSO2 Enterprise Integrator - Registry Database connection information
```
Comment (VimukthiPerera) [inline pattern-2/tile/tile.yml]: **WSO2 Enterprise Integrator** should be changed to **WSO2 API Manager**

File: pattern-2/tile/tile.yml [hunk 7]
```diff
@@ -43,13 +43,38 @@ forms:
...
+  - name: reg_db_username
     type: string
     label: Username
-  - name: carbon_db_password
+  - name: reg_db_password
+    type: secret
+    label: Password
+- name: um_db
+  label: UM DB connection information
+  description: User Manager Database connection information
```
Comment (VimukthiPerera) [inline pattern-2/tile/tile.yml]: The DB description should be **User Management Database connection information**

File: pattern-1/tile/tile.yml [hunk 8]
```diff
@@ -43,13 +43,38 @@ forms:
...
+  - name: reg_db_username
     type: string
     label: Username
-  - name: carbon_db_password
+  - name: reg_db_password
+    type: secret
+    label: Password
+- name: um_db
+  label: UM DB connection information
+  description: User Manager Database connection information
```
Comment (VimukthiPerera) [inline pattern-1/tile/tile.yml]: The description should be **User Management Database connection information**.

File: pattern-1/tile/tile.yml [hunk 9]
```diff
@@ -22,19 +22,19 @@ label: WSO2 API Manager 2.6.0
...
-  version: '3541'
+  version: '170.15'
 
 forms:
-- name: carbon_db
-  label: Carbon DB connection information
-  description: WSO2 API Manager - Carbon Database connection information
+- name: reg_db
+  label: Registry DB connection information
+  description: WSO2 Enterprise Integrator - Registry Database connection information
```
Comment (VimukthiPerera) [inline pattern-1/tile/tile.yml]: The description should be **WSO2 API Manager - Registry Database connection information**

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb (hunk 3 above)
Comment (VimukthiPerera) [reply pattern-2/bosh-release/jobs/wso2apim/templates/ctl.erb:68]: +1

**Review by VimukthiPerera — APPROVED** [review]

**Pull request merged**